raiden/utils/timeout.py

```diff
@@ -0,0 +1,44 @@
+import gevent
+
+from raiden.raiden_service import RaidenService
+from raiden.utils.typing import BlockNumber, Callable
+from raiden.waiting import wait_for_block
+
+
+def _timeout_task(
+    throw: Callable,
+    exception_to_throw: Exception,
+    raiden: RaidenService,
+    block_number: BlockNumber,
+    retry_timeout: float,
+):
+    wait_for_block(raiden, block_number, retry_timeout)
+    throw(exception_to_throw)
+
+
+class BlockTimeout:
```

Comment:
Inpired by `gevent.Timeout`, used to timeout on the settle call (which needs time in blocks and not clock time)